tools/osv-linter/internal/pkgchecker/ecosystems.go

```diff
@@ -2,26 +2,34 @@ package pkgchecker
 
 import (
 	"fmt"
-	"io"
-	"net/http"
-	"regexp"
-	"slices"
-	"strings"
-
-	"golang.org/x/mod/module"
-	"golang.org/x/mod/semver"
-
-	"github.com/ossf/osv-schema/linter/internal/faulttolerant"
-
-	pep440 "github.com/aquasecurity/go-pep440-version"
-
-	"github.com/tidwall/gjson"
 )
 
 // Ecosystem support is a work in progress.
 var SupportedEcosystems = []string{
 	"Go",
 	"PyPI",
+	"crates.io",
+	"npm",
+	"NuGet",
+	"RubyGems",
+	"Packagist",
+	"Pub",
+	"Hackage",
+	"Maven",
+}
+
+// EcosystemBaseURLs maps ecosystems to their base API URLs.
+var EcosystemBaseURLs = map[string]string{
+	"Go":        "https://proxy.golang.org",
+	"PyPI":      "https://pypi.org/pypi",
+	"crates.io": "https://crates.io/api/v1/crates",
+	"npm":       "https://registry.npmjs.org",
+	"NuGet":     "https://api.nuget.org/v3-flatcontainer",
+	"RubyGems":  "https://rubygems.org/api/v1/gems",
+	"Packagist": "https://repo.packagist.org/p2",
+	"Pub":       "https://pub.dev/api/packages",
+	"Hackage":   "https://hackage.haskell.org/package",
+	"Maven":     "https://search.maven.org/solrsearch/select",
```

Comment:
early initial comment: would it make sense to use deps.dev where it's available for an ecosystem? 

some of these have rate limits etc that might be problematic.

Reply:
Yes, I thought about using deps.dev, but I was concerned about the comprehensiveness of its data. So originally I planned to use deps.dev or another source as a secondary check when the primary source failed with 500 errors. But yeah, rate limits are quite important. I will update code to use deps.dev as the primary check source and set the ecosystem-specific registries as secondary ones to bypass the rate limit issue.